### DISCUSSION TITLE
Passwordless connection string for SqlServer

### DISCUSSION THREAD

**Discussion by HiradRoshandel** [quartznet/quartznet#1488] [Q&A] [answered]
In all of the examples where Quartz is connecting to SQL Server I noticed the connection string has both username and password in the string. Is there a way to get Quartz to work with passwordless connection string? I'm using Azure managed identity for my sql server connection. Thanks

**Comment by lahma** [comment, marked as answer]
Quartz just uses ADO.NET connection string you provide, some googling might help with the issue, I found this with quick search: 

* https://colinsalmcorner.com/configuring-aad-authentication-to-azure-sql-databases/
* https://codemag.com/Article/2107041/Eliminate-Secrets-from-Your-Applications-with-Azure-Managed-Identity

**Reply by HiradRoshandel** [reply]
Thanks for your reply @lahma. Maybe I didn't explain well, I understand how to connect to sql server using ADD token however, I'm not sure how I can pass the token to Quartz. The connection string that I'm passing to `properties["quartz.dataSource.default.connectionString"] ` does not include database user or password therefore without the token it wouldn't work.

**Reply by lahma** [reply]
Well how would you pass it from a console app when using `new SqlConnection(theConnectionString)`? I don't think this is Quartz specific?

**Reply by HiradRoshandel** [reply]
@lahma this is how I'm doing it:
```
using Azure.Core;
using Azure.Identity;
using Microsoft.Data.SqlClient;

sqlConnection = new SqlConnection(sqlConnectionString);
var credential = new DefaultAzureCredential();
var token = credential.GetToken(new TokenRequestContext(new[] { "https://database.windows.net" }));
sqlConnection.AccessToken = token.Token;
sqlConnection.Open();
```

**Reply by lahma** [reply]
Hmm.. I think we've actually covered this before already: https://github.com/quartznet/quartznet/discussions/1036

**Reply by HiradRoshandel** [reply]
Thank you I'll check it out. Looks look what I'm looking for.
